packages/ui-react/src/components/Sidebar/Sidebar.tsx

```diff
@@ -16,29 +16,44 @@ type SidebarProps = {
 
 const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, children }) => {
   const { t } = useTranslation('menu');
+  const lastFocusedElementRef = useRef<HTMLElement | null>(null);
+  const sidebarRef = useRef<HTMLDivElement>() as React.MutableRefObject<HTMLDivElement>;
```

Comment:
Setting the default value to `null` should make the ref mutable. It probably wants you to add an optional chain to L26 because the ref might be `null`.
 
```suggestion
  const sidebarRef = useRef<HTMLDivElement>(null);
```

Reply:
Do we have/need a similar functionality for modals? If so, we might want to create a hook for this:

```ts 
export const useModalFocus = (modalRef: MutableRefObject<HTMLElement>, isOpen: boolean) => {
  const lastFocusedElementRef = useRef<HTMLElement | null>(null);

  useEffect(() => {
    // ...
  }, [isOpen]);
};
```

Reply:
done thanks

Reply:
A new comment appeared when I replied "done thanks". Ill write the hook down for the next occurence when i need to set focus back to the last element.